sklearn/compose/tests/test_column_transformer.py

```diff
@@ -571,3 +600,129 @@ def test_column_transformer_remainder_pandas(key):
                            remainder='passthrough')
     assert_array_equal(ct.fit_transform(X_df), X_res_both)
     assert_array_equal(ct.fit(X_df).transform(X_df), X_res_both)
+    assert len(ct.transformers_) == 2
+
+
+@pytest.mark.parametrize("key", [[0], np.array([0]), slice(0, 1),
+                                 np.array([True, False, False])])
+def test_column_transformer_remainder_transformer(key):
+    X_array = np.array([[0, 1, 2],
+                        [2, 4, 6],
+                        [8, 6, 4]]).T
+    X_res_both = X_array.copy()
+
+    # second and third columns are doubled when remainder = DoubleTrans
+    X_res_both[:, 1:3] *= 2
+
+    ct = ColumnTransformer([('trans1', Trans(), key)],
+                           remainder=DoubleTrans())
+
+    assert_array_equal(ct.fit_transform(X_array), X_res_both)
+    assert_array_equal(ct.fit(X_array).transform(X_array), X_res_both)
+    assert len(ct.transformers_) == 2
+
+
+def test_column_transformer_drops_all_remainder_transformer():
+    X_array = np.array([[0, 1, 2],
+                        [2, 4, 6],
+                        [8, 6, 4]]).T
+
+    # columns are doubled when remainder = DoubleTrans
+    X_res_both = 2 * X_array.copy()[:, 1:3]
+
+    ct = ColumnTransformer([('trans1', 'drop', [0])],
+                           remainder=DoubleTrans())
+
+    assert_array_equal(ct.fit_transform(X_array), X_res_both)
+    assert_array_equal(ct.fit(X_array).transform(X_array), X_res_both)
+    assert len(ct.transformers_) == 2
+
+
+def test_column_transformer_sparse_remainder_transformer():
+    X_array = np.array([[0, 1, 2],
+                        [2, 4, 6],
+                        [8, 6, 4]]).T
+
+    ct = ColumnTransformer([('trans1', Trans(), [0])],
+                           remainder=SparseMatrixTrans())
+
+    X_trans = ct.fit_transform(X_array)
+    assert_true(sparse.issparse(X_trans))
```

Comment:
With the adoption of pytest, we are phasing out use of test helpers `assert_equal`, `assert_true`, etc. Please use bare `assert` statements, e.g. `assert x == y`, `assert not x`, etc.

You also don't need `assert_dict_equal`, just `assert d1 == d2`.

Reply:
I have a habit of keeping to the same style as the file I am working with. 

Since I am changing the new tests to bare asserts, do you mind if I go through the whole test file and change everything to bare asserts?

Reply:
To avoid making it harder to review, don't do it here. I don't mind this coming in a separate PR, given that there isn't yet much concurrent work on this file. Don't replace the array comparison assertions. Mostly the assert_true/false/equal are worth removing